### PR TITLE
Bug 1997108: fix react warnings while loading topology page

### PR DESCRIPTION
**Pull request by sahil143**
fixes: https://issues.redhat.com/browse/ODC-6263

This PR fixes react warnings on the topology page with setQueryArguments and setState call.

### CODE REVIEW

File: frontend/packages/topology/src/components/graph-view/Topology.tsx [hunk 1]
```diff
@@ -186,14 +186,24 @@ const Topology: React.FC<TopologyProps &
...
-    visualizationRef.current = createVisualization();
-  }
   const visualization = visualizationRef.current;
 
+  React.useEffect(() => {
+    if (!visualizationRef.current) {
+      visualizationRef.current = createVisualization();
+    }
+  }, [createVisualization]);
+
```
Comment (christianvogt) [inline frontend/packages/topology/src/components/graph-view/Topology.tsx:206]: Updating a `ref` in the render flow is ok. This doesn't require a change for it to be done inside an effect.

Comment (sahil143) [reply frontend/packages/topology/src/components/graph-view/Topology.tsx:206]: there is a setState call inside `createVisualization`, that's why moved the ref inside an effect.

Comment (christianvogt) [reply frontend/packages/topology/src/components/graph-view/Topology.tsx:206]: I didn't realize that `createVisualization` was executing a callback which could set state. In that case what about:
```
  if (!visualizationRef.current) {
    visualizationRef.current = createVisualization();
  }
  const visualization = visualizationRef.current;
  React.useEffect(() => {
    if (visualization) {
      setVisualization(visualization);
    }
  }, [visualization]);
```

Comment (christianvogt) [reply frontend/packages/topology/src/components/graph-view/Topology.tsx:206]: Make sure to also fix-up `TopologyListView` which does something similar.

**Comment by invincibleJai** [comment]
/approve

cc @christianvogt

File: frontend/packages/topology/src/components/graph-view/Topology.tsx (hunk 1 above)
Comment (sahil143) [reply frontend/packages/topology/src/components/graph-view/Topology.tsx]: @christianvogt fixed

File: frontend/packages/dev-console/src/plugin.tsx [hunk 2]
```diff
@@ -95,10 +95,10 @@ const plugin: Plugin<ConsumedExtensions> = [
...
         '/topology/all-namespaces',
         '/topology/ns/:name',
-        '/topology/all-namespaces/graph',
-        '/topology/ns/:name/graph',
-        '/topology/all-namespaces/list',
-        '/topology/ns/:name/list',
+        '/topology/all-namespaces?view=graph',
+        '/topology/ns/:name?view=graph',
+        '/topology/all-namespaces?view=list',
+        '/topology/ns/:name?view=list',
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/plugin.tsx:101]: I think you can remove these paths altogether since the query params don't matter here.

Comment (sahil143) [reply frontend/packages/dev-console/src/plugin.tsx]: removed, only kept ones that are without the query params

**Comment by sahil143** [comment]
@makambalaji The tests are failing on this PR as well for helm.

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

File: frontend/packages/dev-console/src/plugin.tsx (hunk 2 above)
Comment (cardil) [reply frontend/packages/dev-console/src/plugin.tsx]: Can I presume this change is OCP 4.9+ only?

Comment (christianvogt) [reply frontend/packages/dev-console/src/plugin.tsx]: Yes that's correct.

**Pull request merged**